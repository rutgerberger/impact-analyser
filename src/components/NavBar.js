import React from "react";
import { Link, BrowserRouter as Router } from "react-router-dom";

function NavBar({ }) {
    return(
        <nav className='navbar'>
            <Router> <Link className='nav-button' to='/'>
                Impact Analyser
            </Link></Router>
            <Router><Link className='nav-button' to='/'>
                Contact
            </Link></Router>
        </nav> 
    );
}

export default NavBar;
