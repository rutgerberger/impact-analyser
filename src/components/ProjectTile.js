import React from 'react'

const ProjectTile = ({ name, id, mainContributor, totalHours, contributors }) => {
    return (
      <div>
        <div className="left-tile">
          <ul>
          <li>
          Name:
          </li>
          <li>
          ID:
          </li>
          <li>
          Main Contributor:
          </li>
          <li>
          Expected Hours of Work:
          </li>
          <li>
          No. of contributors:
          </li>
          </ul>
        </div>
        <div className="project-tile">
          <ul>
          <li className="project-tile-title">
          {name}
          </li>
          <li>
          {id}
          </li>
          <li>
          {mainContributor}
          </li>
          <li>
          {totalHours}
          </li>
          <li>
          {contributors}
          </li>
          </ul>
        </div>
      </div>
    )
  }

export default ProjectTile