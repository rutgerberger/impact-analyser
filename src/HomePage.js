import React from 'react'
import NavBar from './components/NavBar'
import { Link, BrowserRouter as Router } from "react-router-dom";

function HomePage() {

    return(
      <div className="styling">
        <NavBar />
        <div className='header'>
               <h2>IMPACT ANALYSER</h2>
               <p>MAKING TASK DIVIDING EASY</p>
        </div>
        <div className='container'>
          <div className='fancy-image'>

          </div>
          <div className= 'content-container'>
              <Router><Link to='/projects'className='button-home'>

                <div>
                  Projects
                </div>

              </Link></Router>
              <Router><Link to='/profile'className='button-home'>

                <div>
                  Profile
                </div>

              </Link></Router>
              <Router><Link to='/about' className='button-home'>

                <div>
                  About
                </div>

              </Link></Router>
          </div>
        </div>
      </div>
    )
}

export default HomePage