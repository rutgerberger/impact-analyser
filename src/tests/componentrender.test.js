import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';
import ProjectTile from  '../components/ProjectTile'
import HomePage from '../HomePage';
test('renders correctly', () => {
  const tree = renderer.create(<ProjectTile />).toJSON();
  expect(tree).toMatchSnapshot();
  const tree2 = renderer.create(<HomePage/>).toJSON();
  expect(tree2).toMatchSnapshot();
  const tree3 = renderer.create(<NavBar/>).toJSON();
  expect(tree3).toMatchSnapshot();
});