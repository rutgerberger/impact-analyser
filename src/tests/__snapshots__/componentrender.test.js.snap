// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="left-tile"
  >
    <ul>
      <li>
        Name:
      </li>
      <li>
        ID:
      </li>
      <li>
        Main Contributor:
      </li>
      <li>
        Expected Hours of Work:
      </li>
      <li>
        No. of contributors:
      </li>
    </ul>
  </div>
  <div
    className="project-tile"
  >
    <ul>
      <li
        className="project-tile-title"
      />
      <li />
      <li />
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="styling"
>
  <nav
    className="navbar"
  >
     
    <a
      className="nav-button"
      href="/"
      onClick={[Function]}
    >
      Impact Analyser
    </a>
    <a
      className="nav-button"
      href="/"
      onClick={[Function]}
    >
      Contact
    </a>
  </nav>
  <div
    className="header"
  >
    <h2>
      IMPACT ANALYSER
    </h2>
    <p>
      MAKING TASK DIVIDING EASY
    </p>
  </div>
  <div
    className="container"
  >
    <div
      className="fancy-image"
    />
    <div
      className="content-container"
    >
      <a
        className="button-home"
        href="/projects"
        onClick={[Function]}
      >
        <div>
          Projects
        </div>
      </a>
      <a
        className="button-home"
        href="/profile"
        onClick={[Function]}
      >
        <div>
          Profile
        </div>
      </a>
      <a
        className="button-home"
        href="/about"
        onClick={[Function]}
      >
        <div>
          About
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<nav
  className="navbar"
>
   
  <a
    className="nav-button"
    href="/"
    onClick={[Function]}
  >
    Impact Analyser
  </a>
  <a
    className="nav-button"
    href="/"
    onClick={[Function]}
  >
    Contact
  </a>
</nav>
`;
